### ISSUE TITLE
Major issue with Release 23.4.0

### ISSUE THREAD

**Issue by ggnaegi** [ThreeMammals/Ocelot#2207]
## Expected Behavior / New Feature

We have internal errors everywhere

## Actual Behavior / Motivation for New Feature

We shouldn't have internal errors everywhere

@raman-m We have a big problem with this release, it doesn't work. I don't know where the problem is, but it's returning us internal errors.

**Comment by raman-m** [comment]
I will write you in Slack.
I'm sad and wondered that the quality of this feature is not the same as we thought and expected.
Please explain and attack the logs with errors!

**Comment by ggnaegi** [comment]
Yes it's a sad thing because we have foreseen a lot of tests. The hyphen can't be used as regex group name.

**Comment by raman-m** [comment]
Gui, please decide! &rarr; Would you prefer to leave this issue as a bug, or convert to feat?
My opinion we don't have a bug at all! But... we have a problem with lack of instructions in Placeholders docs where we must explain to users what characters can be used in placeholder name. Now we support only alphanumeric chars only, with underscore char `_`. Also users can use casing aka upper chars, as far as I know, placeholder name is case sensitive.

**Comment by ggnaegi** [comment]
@raman-m I will have a more in-depth analysis of the situation. I think, we should first wait for more feedbacks. In my case, we have decided to use camelCasing instead.

**Comment by raman-m** [comment]
> In my case, we have decided to use Camel Casing instead.

Good, clever boys! In addition to camel case names, we can throw in digits and `_` (underscore). These cover 99.99999999999% of developers, clients, and customers. But you, my friend, are lucky: you've got a deal with that elusive 0.0000000001% of customers! 🎉 🤣 

> I think, we should first wait for more feedbacks.

This issue should be opened, correct?
Additionally, should we updating the documentation to clarify allowed characters?

**Comment by ggnaegi** [comment]
> This issue should be opened, correct? Additionally, should we updating the documentation to clarify allowed characters?

Yes and Yes

**Comment by ggnaegi** [comment]
By the way, standard behavior for routing in .NET -> word with hyphens, camelCase, eg. user-id -> userId. So maybe we should follow this logic, but it's not a bug, it's a feature.

**Comment by raman-m** [comment]
> standard behavior for routing in .NET -> word with hyphens, camelCase, eg. user-id -> userId.

@ggnaegi  Please provide the links to the MS Learn documentation. I have not heard about this routing feature.

**Comment by raman-m** [comment]
In my opinion, we need to update the Ocelot documentation to explain naming rules for the [Placeholders](https://ocelot.readthedocs.io/en/latest/features/routing.html#placeholders).
Therefore, I have marked this issue with the `documentation` label.

**Comment by raman-m** [comment]
> @fmattioli [2 weeks ago](https://github.com/ThreeMammals/Ocelot/issues/2207#issuecomment-2786587625)

Well, Stack Overflow is not an official source of documentation.
Kindly provide links exclusively to the Microsoft Learn documentation portal.

**Comment by raman-m** [comment]
> @ggnaegi [on Nov 19, 2024](https://github.com/ThreeMammals/Ocelot/issues/2207#issuecomment-2485637635)

Will you work on this?
How is it critical for you and for the community? I see 1 feedback only from the community...
